cmd/testrunner/sensei.go

```diff
@@ -0,0 +1,243 @@
+	err = tryFunc(
+		func() error {
+			var err error
+			conn, err = grpc.Dial(rpcHost, opts...)
+			if err != nil {
+				return err
+			}
+			return err
+		}, 60)
+	if err != nil {
+		return nil, err
+	}
+
+	return &senseiConnection{
+		alias:  alias,
+		conn:   conn,
+		pubKey: pubkey,
+		client: sensei.NewNodeClient(conn),
+	}, nil
+}
+
+func (l *senseiConnection) PubKey() string {
+	return l.pubKey.String()
+}
+
+func (l *senseiConnection) Close() {
+	l.conn.Close()
+}
+
+func (l *senseiConnection) Connect(key, host string) error {
+	_, err := l.client.ConnectPeer(context.Background(), &sensei.ConnectPeerRequest{
+		NodeConnectionString: fmt.Sprintf("%v@%v", key, host),
+	})
+
+	return err
+}
+
+func (l *senseiConnection) NewAddress() (string, error) {
+	resp, err := l.client.GetUnusedAddress(
+		context.Background(),
+		&sensei.GetUnusedAddressRequest{},
+	)
+	if err != nil {
+		return "", err
+	}
+	return resp.Address, nil
+}
+
+func (l *senseiConnection) OpenChannel(peerKey string, amtSat int64,
+	pushAmtSat int64, private bool) error {
+
+	_, err := l.client.OpenChannels(context.Background(), &sensei.OpenChannelsRequest{
+		Channels: []*sensei.OpenChannelInfo{
+			{
+				AmtSatoshis: uint64(amtSat),
+				Public:      !private,
+
+				// TODO:
+				// NodePubkeyString:   peerKey,
+				// SpendUnconfirmed:   true,
+				// PushSat:            pushAmtSat,
+			},
+		},
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (l *senseiConnection) SetPolicy(chanPoint *lnrpc.ChannelPoint,
+	policy *graphreader.PolicyData) error {
+
+	// Not necessary for sender node.
+	panic("not implemented")
+}
+
+func (l *senseiConnection) ActiveChannels() (int, error) {
+	resp, err := l.client.ListChannels(context.Background(), &sensei.ListChannelsRequest{})
+	if err != nil {
+		return 0, err
+	}
+
+	// TODO: How to get numer of active channels?
+	return len(resp.Channels), nil
+}
+
+func (l *senseiConnection) IsSynced(totalEdges, localChannels int) (bool, error) {
+	// TODO: How to get number of edges received through gossip + local edges?
+
+	panic("not implemented")
+}
+
+func (l *senseiConnection) AddInvoice(amtMsat int64) (string, error) {
+	// Not necessary for sender node.
+	panic("not implemented")
+}
+
+func (l *senseiConnection) SendPayment(invoice string, aliasMap map[string]string) error {
+	panic("not implemented")
+}
+
+func (l *senseiConnection) HasFunds() error {
+	for {
+		resp, err := l.client.GetBalance(context.Background(),
+			&sensei.GetBalanceRequest{})
+		if err != nil {
+			return err
+		}
+
+		// TODO: Is confirmed balance only?
+		if resp.BalanceSatoshis > 0 {
+			return nil
+		}
+
+		time.Sleep(time.Second)
+	}
+}
+
+func (l *senseiConnection) Restart() (nodeInterface, error) {
+	// Not needed for sensei to get synced correctly - hopefully.
+	panic("not implemented")
+}
+
+func (l *senseiConnection) GetChannelBalance() (int, error) {
+	// TODO: Add up list channel responses?
+	panic("not implemented")
```

Comment:
It should now be included in the GetBalance response mentioned above.  You could also manually add up the fields of list channel depending on the exact calculation of balance you want.